Dockerfile

```diff
@@ -0,0 +1,100 @@
+#  Copyright (C) 2018 LEIDOS.
+# 
+#  Licensed under the Apache License, Version 2.0 (the "License"); you may not
+#  use this file except in compliance with the License. You may obtain a copy of
+#  the License at
+# 
+#  http://www.apache.org/licenses/LICENSE-2.0
+# 
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#  License for the specific language governing permissions and limitations under
+#  the License.
+
+# CARMA Docker Configuration Script
+#
+# Performs all necessary tasks related to generation of a generic CARMA docker image
+# suitable for deployment to a vehicle. The generic image will still need to be 
+# configured for each vehicle by means of volume mapping configuration files and
+# networking mapping at run time
+#
+# Build Arguments:
+# SSH_PRIVATE_KEY - If the extra package repository to be used during the build requires
+#                   authentication, please pass in the necessary SSH private key in text
+#                   form here, most likely via "$(cat ~/.ssh/id_rsa)". This data is not 
```

Comment:
Is it possible to have the user just pass in ~/.ssh/id_rsa and the script would apply the $(cat )

Reply:
I can give it a quick test, but I don't know if that's possible. The docker build process actually runs as a daemon under it's own user, so it shouldn't have access to `~/.ssh/id_rsa` normally as that's `600` permissions.

Reply:
Actually, testing it out, the docker build process only has access to files at or below the Dockerfile in the folder structure, so the build process can't even see the user's home folder. `cat`ing the key is kind of a work around hack I guess